What's New 115 (#2654)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/whats-new/2023/07/devtools-115.md

---
title: What's new in DevTools (Microsoft Edge 115)
description: "Improvements for connecting DevTools to Azure Artifacts symbol server. Snapshotting improvements for the Memory tool. Refreshing the DevTools icons and buttons. And more."
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 08/18/2023
---
# What's New in DevTools (Microsoft Edge 115)

[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]

<!-- ====================================================================== -->
## Improvements for connecting DevTools to Azure Artifacts symbol server

<!-- Subtitle: You can now use Azure Active Directory for authentication instead of Personal Access Tokens (PATs). You can connect to multiple symbol servers and filter which sourcemaps you want DevTools to fetch. -->

> In Microsoft Edge 115, it's easier and more customizable to connect DevTools to Azure Artifacts symbol server.  In previous versions of Microsoft Edge, to authenticate to the symbol server, you had to generate Personal Access Tokens (PATs) in Azure DevOps and then paste them into DevTools. Now, as long as you're logged into Microsoft Edge, you can connect to the symbol server by using Azure Active Directory.

Reviewer: I believe this requires a business account. I don't believe `as long as you're logged in` is adequate. I don't know what the proper name is for this, however.

Author: @robpaveza - probably "work account", not "business account", based on full-text search of this repo.

Author: or maybe should be:
using a Microsoft account (MSA) that's a work account

Author: replaced this comment thread by fresh comment https://github.com/MicrosoftDocs/edge-developer/pull/2654/files#r1303525933

> Additionally, you can now connect to multiple symbol servers, in case you need to fetch sourcemaps from different Azure DevOps organizations. You can also filter which sourcemaps you want DevTools to fetch from the symbol server, by specifying the sourcemap URLs in an inclusion or exclusion list.

Reviewer: It might be worthwhile to specify what the wildcards match:

>
>  * `?` will match a single character
>  * `*` will match one or more of any character
> 
> "Inclusion list" mode will only attempt to look up source maps for scripts with URLs that match one of the entries in the list. "Exclusion list" mode will attempt to look up any source maps for scripts *except* those with URLs that match one of the entries in the list.
>
> An example might be to do something like:
> ```https://cdn.contoso.com/*```
> ```https://packages.contoso.com/*```
> on an Inclusion list will only attempt to resolve source maps matching these two URL patterns, and won't attempt to load source maps for other scripts.
>
> The Source Maps Monitor tool has been updated to show what has or has not been attempted.

Author: @robpaveza - in my Draft 2, i wish to be more specific than:
* "it attempts" (what is "it"?), 
* "it only attempts" (what is "it")? and 
* "what hasn't been attempted"

Draft 2:

In the **Filter behavior** text box, the following wildcards are supported:

| Wildcard | Meaning |
|---|---|
| ? | Matches a single character. |
| * | Matches one or more of any character. |

If you select **Exclusion List**, it attempts to look up any source maps for scripts except those that have a URL that matches one of the entries in the list.

If you select **Inclusion List**, it only attempts to look up source maps for scripts that have URLs that match one of the entries in the list.  For example, suppose you select **Inclusion List** in the **Filter behavior** dropdown list, and then enter the following in the **Filter behavior** text box:

```http
https://cdn.contoso.com/*
https://packages.contoso.com/*
```

In this example, it only attempts to resolve source maps that match these two URL patterns, and doesn't attempt to load source maps for other scripts.  The [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) has been updated to show what has or hasn't been attempted.

Author: i pushed that commit and entered fresh comments to fine-tune it

## Screen readers correctly announce dropdown state in Device Mode

<!-- Subtitle: In Device Mode, screen readers now correctly announce the state of dropdown menus (for example, announcing "collapsed" when the dropdown menu is collapsed). -->

In previous versions of Microsoft Edge, in Device Mode, screen readers incorrectly announced the state of dropdown menus as "expanded" when these menus were actually collapsed.  In Microsoft Edge 115, this issue has been fixed and screen readers now correctly announce the state of dropdown menus in Device Mode as "collapsed" or "expanded":

![The 'Throttling' dropdown menu expanded in Device Mode](./devtools-115-images/device-mode-dropdown.png)

See also:
* [Emulate mobile devices (Device Emulation)](../../../device-mode/index.md)

<!-- ====================================================================== -->
## Profiling improvements for the Performance tool

<!-- Subtitle: Your web content will now be more responsive while profiling in the Performance tool, and "Profiler Overhead" blocks of CPU time have been added, to account for the additional time Microsoft Edge needs when profiling. -->

> When recording a profile in the **Performance** tool, the V8 JavaScript engine takes a sample of all the JavaScript functions that are running, to accurately report CPU time in the flame chart.  When profiling a website that has many cross-domain iframes, these samples take a long time to collect.  This results in significant overhead, slowing down the main thread, which slows down the performance of the website while profiling.  In Microsoft Edge 115 on Windows, V8 collects these samples more efficiently, resulting in better site performance while profiling.

Author: ```suggestion
When recording a profile in the **Performance** tool, the V8 JavaScript engine takes periodic samples of all the JavaScript functions that are running, to accurately report CPU time in the flame chart.  When profiling a website that has many cross-domain iframes, these samples take a long time to collect.  This results in significant overhead, slowing down the main thread, which slows down the performance of the website while profiling.  In Microsoft Edge 115 on Windows, V8 collects these samples more efficiently, resulting in better site performance while profiling.
```

Author: done locally

---
title: What's new in DevTools (Microsoft Edge 115)
description: "Improvements for connecting DevTools to Azure Artifacts symbol server. Snapshotting improvements for the Memory tool. Refreshing the DevTools icons and buttons. And more."
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 08/18/2023
---
# What's New in DevTools (Microsoft Edge 115)

[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]

<!-- ====================================================================== -->
## Improvements for connecting DevTools to Azure Artifacts symbol server

<!-- Subtitle: You can now use Azure Active Directory for authentication instead of Personal Access Tokens (PATs). You can connect to multiple symbol servers and filter which sourcemaps you want DevTools to fetch. -->

> In Microsoft Edge 115, it's easier and more customizable to connect DevTools to Azure Artifacts symbol server.  In previous versions of Microsoft Edge, to authenticate to the symbol server, you had to generate Personal Access Tokens (PATs) in Azure DevOps and then paste them into DevTools. Now, as long as you're logged into Microsoft Edge using a work account, you can connect to the symbol server by using Azure Active Directory.

Author: ```suggestion
In Microsoft Edge 115, it's easier and more customizable to connect DevTools to Azure Artifacts symbol server.  In previous versions of Microsoft Edge, to authenticate to the symbol server, you had to generate Personal Access Tokens (PATs) in Azure DevOps and then paste them into DevTools. Now, as long as you're logged into Microsoft Edge using a work account, you can connect to the symbol server that uses Azure Active Directory.
```
@zoherghadyali - tech review this change.  orig wording was:
connect to the symbol server with Azure Active Directory.

https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/u/using-by-using-with

Author: keep as "by using"

> <!-- todo: at "work account", specify eg: 

Author: @robpaveza - is "work account" sufficient, or should it specify eg:

a Microsoft account (MSA) that's a work account

enterprise customers who are signed into Microsoft Edge with a Microsoft Azure Active Directory (AAD) account

Reviewer: I think "a work account" is sufficient if that's a thing that Edge has generally. I think the internal differentiator is "MSA" which is personal, and "AAD", which is for business/enterprise, so perhaps AAD is the best version.

Author: retained visible wording.  revised hidden comment as:
```
<!-- re: work account, possibly:
enterprise customers who are signed into Microsoft Edge with a Microsoft Azure Active Directory (AAD) account
a Microsoft account (MSA) that's a work account
-->
```

a Microsoft account (MSA) that's a work account
enterprise customers who are signed into Microsoft Edge with a Microsoft Azure Active Directory (AAD) account
-->

Additionally, you can now connect to multiple symbol servers, in case you need to fetch sourcemaps from different Azure DevOps organizations. You can also filter which sourcemaps you want DevTools to fetch from the symbol server, by specifying the sourcemap URLs in an inclusion or exclusion list.

In the **Filter behavior** text box, the following wildcards are supported:

| Wildcard | Meaning |
|---|---|
| ? | Matches a single character. |
| * | Matches one or more of any character. |

> If you select **Exclusion List**, it attempts to look up any source maps for scripts except those that have a URL that matches one of the entries in the list.

Author: ```suggestion
If you select **Exclusion List**, the symbol server attempts to look up any source maps for scripts except those that have a URL that matches one of the entries in the list.
```
@robpaveza - tech review this change, i'm trying to be more specific than "it"

Reviewer: I think the preference here is "DevTools," as in *DevTools attempts to look up..*.

Author: DevTools attempts to look up

> If you select **Inclusion List**, it only attempts to look up source maps for scripts that have URLs that match one of the entries in the list.  For example, suppose you select **Inclusion List** in the **Filter behavior** dropdown list, and then enter the following in the **Filter behavior** text box:

Author: ```suggestion
If you select **Inclusion List**, it only attempts to look up source maps for scripts that have URLs that match one of the entries in the list.  For example, suppose you select **Inclusion List** in the **Filter behavior** dropdown list, and then enter the following in the **Filter behavior** text box:
```
@robpaveza - tech review this other change, i'm trying to be more specific than "it"

Reviewer: It's really "DevTools", as in, *DevTools only attempts to look up source maps*.

Author: DevTools only attempts to look up source maps ...

```http
https://cdn.contoso.com/*
https://packages.contoso.com/*
```

> In this example, it only attempts to resolve source maps that match these two URL patterns, and doesn't attempt to load source maps for other scripts.  The [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) has been updated to show what has or hasn't been attempted.

Author: ```suggestion
In this example, the symbol server only attempts to resolve source maps that match these two URL patterns, and doesn't attempt to load source maps for other scripts.  The [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) has been updated to show which URL lookups have or haven't been attempted.
```
@robpaveza - tech review these two changes, i'm trying to be more specific

Reviewer: I think in the former case, we're again looking at "DevTools": *In this example, DevTools only attempts...*.

In the latter case, I think your proposed change is fine.

Author: In this example, DevTools only attempts to resolve ...

The [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) has been updated to show which URL lookups have or haven't been attempted.

#### Wildcards

In the **Filter behavior** text box, the following wildcards are supported:

| Wildcard | Meaning |
|---|---|
| ? | Matches a single character. |
| * | Matches one or more of any character. |

If you select **Exclusion List**, DevTools attempts to look up any source maps for scripts except those that have a URL that matches one of the entries in the list.

If you select **Inclusion List**, DevTools only attempts to look up source maps for scripts that have URLs that match one of the entries in the list.  For example, suppose you select **Inclusion List** in the **Filter behavior** dropdown list, and then enter the following in the **Filter behavior** text box:

```http
https://cdn.contoso.com/*
https://packages.contoso.com/*
```

In this example, DevTools only attempts to resolve source maps that match these two URL patterns, and doesn't attempt to load source maps for other scripts.

<!-- ------------------------------ -->
#### Source Maps Monitor tool

The [Source Maps Monitor tool](../../../source-maps-monitor/source-maps-monitor-tool.md) has also been updated, to show which URL lookups have or haven't been attempted.  For example:

> 

Reviewer 2: I think the section about the Source Maps Monitor tool should be cut and we should just take a work item to update the Source Maps Monitor tool with new screenshots. These instructions don't make sense because our readers won't have the **microsoft** org to select.

1. In DevTools **Settings > Symbol Server**, in the **Authorization mode** dropdown list, select **Azure Active Directory**.

1. Double-click the **microsoft** org to select it.

1. In the **Filter behavior** dropdown list, select **Exclusion List**.

1. In the **Filter behavior** text box, enter a domain with wildcard (`https://ryanseddon.com/*`) to exclude.

1. Close DevTools **Settings**.

As a result, the **Source Maps Monitor** tool indicates excluded source map URL lookup attempts:

![Source Maps Monitor tool indicates source map URL lookups attempted](./devtools-115-images/source-maps-monitor.png)

See also:
* [Securely debug original code by publishing source maps to the Azure Artifacts symbol server](../../../javascript/publish-source-maps-to-azure.md)
* [Securely debug original code by using Azure Artifacts symbol server source maps](../../../javascript/consume-source-maps-from-azure.md)

<!-- ====================================================================== -->

> ## Improvements for the Memory tool

Reviewer 2: This title is too similar to the one immediately preceding it. Can we come up with a different word for "Improvements"
 ```suggestion
## Allocation instrumentation on timeline works better and heap snapshots are now more accurate in the Memory tool
```

Author: `## Memory tool enhancements`

---
title: What's new in DevTools (Microsoft Edge 115)
description: "Improvements for connecting DevTools to Azure Artifacts symbol server. Snapshotting improvements for the Memory tool. Refreshing the DevTools icons and buttons. And more."
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 08/18/2023
---
# What's New in DevTools (Microsoft Edge 115)

[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]

<!-- ====================================================================== -->
## Improvements for connecting DevTools to Azure Artifacts symbol server

<!-- Subtitle: You can now use Azure Active Directory for authentication instead of Personal Access Tokens (PATs). You can connect to multiple symbol servers and filter which sourcemaps you want DevTools to fetch. -->

<!-- Reviewer: Rob Paveza-->

> In Microsoft Edge 115, we've made it easier and more customizable to connect DevTools to Azure Artifacts symbol server.  In previous versions of Microsoft Edge, you had to generate Personal Access Tokens (PATs) in Azure DevOps and paste them into DevTools in order to authenticate to the symbol server. Now, as long as you are logged into Microsoft Edge, you can connect to the symbol server by using<!--todo: that uses? --> Azure Active Directory.

Author: @zoherghadyali -- I changed from "with" to "by using".  Should it instead read "that uses"?

* connect to the symbol server with Azure Active Directory.
* connect to the symbol server by using Azure Active Directory.
* connect to the symbol server that uses Azure Active Directory.

Author: the surrounding Markdown is Outdated

Author: keep "by using"

![Connecting to symbol server using Azure Active Directory from DevTools settings](./devtools-115-images/aad-symbol-server.png)
<!-- todo: steps to re-create screenshot -->
<!--
1. Launch Edge, ensure that you are logged in to the browser
1. Open DevTools > Settings > Symbol Server
1. Under **Authorization mode:**, select the dropdown and select Azure Active Directory
1. Load the Azure DevOps organizations
1. Take a screenshot
1. Draw red highlight boxes around Authorization mode and Filter behavior
1. See https://dev.azure.com/microsoft/Edge/_git/chromium.devtools-frontend/pullrequest/8874508 for more info
-->

See also:
* [Securely debug original code by publishing source maps to the Azure Artifacts symbol server](../../../javascript/publish-source-maps-to-azure.md)
* [Securely debug original code by using Azure Artifacts symbol server source maps](../../../javascript/consume-source-maps-from-azure.md)

<!-- ====================================================================== -->
## Snapshotting improvements for the Memory tool
<!-- 

> ## Improvements to Memory Tool

Author: Change heading?

Author: `## Improvements for the Memory tool`

> * [Building Chat Plugins for Microsoft Bing and Edge](https://www.youtube.com/watch?v=Q-5M7EYjl6U&list=PL4z1-7pjJU6zJT3PBQ4mTbNg2wtX7Lt52)

<!-- ====================================================================== -->
## Entry 1

<!-- Subtitle: . -->

![todo: Alt-text 1](./devtools-115-images/template.png)
<!-- todo: steps to re-create screenshot -->

<!--
See also:
* []()
-->

<!-- ====================================================================== -->
## Entry 2

<!-- Subtitle: . -->

![todo: Alt-text 2](./devtools-115-images/template.png)
<!-- todo: steps to re-create screenshot -->

<!--
See also:
* []()
-->

<!-- ====================================================================== -->
## Entry 3

<!-- Subtitle: . -->

![todo: Alt-text 3](./devtools-115-images/template.png)
<!-- todo: steps to re-create screenshot -->

<!--
See also:
* []()
-->

<!-- ====================================================================== -->
## Announcements from the Chromium project

Microsoft Edge 115 also includes the following updates from the Chromium project:

> <!-- todo: delete some links? -->

Author: @zoherghadyali @jenna-sasson  Which links to remove?  
Rendered article section:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/whats-new/2023/07/devtools-115?branch=pr-en-us-2654#announcements-from-the-chromium-project

Author: links removed
